### PR TITLE
LPS-161572 Break the circular dependency between CPDefinitionLocalSer…

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-161572

Hi @tinatian @dantewang

This pull breaks the circular dependency between CPDefinitionLocalService and CPDisplayLayoutLocalServiceby using persistence class directly.

Lily

### CODE REVIEW

**Comment by slnn** [comment]
Hi @zhangyan0701

Could you help to scan test results when they are finished?

Thanks
Lily

**Comment by zhangyan0701** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -1005,8 +1005,14 @@ public CPDefinition deleteCPDefinition(CPDefinition cpDefinition)
...
 
-		cpDisplayLayoutLocalService.deleteCPDisplayLayouts(
-			CPDefinition.class, cpDefinition.getCPDefinitionId());
+		List<CPDisplayLayout> cpDisplayLayouts =
+			cpDisplayLayoutPersistence.findByC_C(
+				classNameLocalService.getClassNameId(CPDefinition.class),
+				cpDefinition.getCPDefinitionId());
+
+		for (CPDisplayLayout cpDisplayLayout : cpDisplayLayouts) {
+			cpDisplayLayoutPersistence.remove(cpDisplayLayout);
```
Comment (dantewang) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1014]: @slnn The original code calls `CPDisplayLayoutLocalService.deleteCPDisplayLayout(CPDisplayLayout)` which has the `@Indexable` annotation. When you switch it to directly use the persistence, the index will not be updated.

```	@Indexable(type = IndexableType.DELETE)
	@Override
	public CPDisplayLayout deleteCPDisplayLayout(
		CPDisplayLayout cpDisplayLayout) {

		return cpDisplayLayoutPersistence.remove(cpDisplayLayout);
	}
```

Comment (dantewang) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1014]: @slnn You may need to manually remove the indexes of `CPDisplayLayout` or find another way to break the circular dependency.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (dantewang) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1014]: cc @tinatian

**Comment by slnn** [comment]
ci:test:sf

**Comment by slnn** [comment]
ci:test:relevant

**Comment by slnn** [comment]
ci:test:commerce

**Comment by slnn** [comment]
Hi @zhangyan0701 

Could you help to scan test results when they are finished?

Thanks
Lily

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (slnn) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1014]: Hi @dantewang 

I add the index process manually in the last commit after persistence's remove method.

Lily

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by slnn** [comment]
Hi @dantewang

Since the change for disabling auto import ([LPS-161869](https://issues.liferay.com/browse/LPS-161869)) has merged into master, I rebase this branch with the master and forced push on this pull.

Lily

**Comment by slnn** [comment]
With the forced push change, the ci test has been ran in:https://github.com/slnn/liferay-portal/pull/631

**Comment by tinatian** [comment]
@slnn Code looks good to me, wait for commerce team to review.

Tina

**Comment by victoroscuro** [comment]
ci:test:commerce

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina.

**Pull request closed without merging**